### PR TITLE
util/admin: migrate admin test to testify

### LINKED ISSUE #26102: migrate test-infra to testify for `util/admin` pkg

### PR DESCRIPTION
**Pull request by evilbinary**
### What problem does this PR solve?

Issue Number: close #26102

### What is changed ?

Migrate test-infra to testify for admin pkg.

### Release note

- No release note

### CODE REVIEW

**Comment by tisonkun** [comment]
This PR seems in a quite different code style comparing with other migration patches. I've created a sibling version which you can make use of https://github.com/tisonkun/tidb/commit/eb35f5c901fa4868edfe3bec8c4c858687f37ee8 . We should keep consistency when coding IMO.

**Comment by tisonkun** [comment]
Also please sign the CLA for the process proceed.

File: util/admin/admin_test.go [hunk 1]
```diff
@@ -389,6 +372,44 @@ func (s *testSuite) TestError(c *C) {
 	}
 	for _, err := range kvErrs {
 		code := terror.ToSQLError(err).Code
-		c.Assert(code != mysql.ErrUnknown && code == uint16(err.Code()), IsTrue, Commentf("err: %v", err))
+		require.NotEqual(t, mysql.ErrUnknown, code)
+		require.Equal(t, uint16(err.Code()), code)
+	}
+}
+
+func newMockStore(t *testing.T) (store kv.Storage, clean func()) {
```
Comment (xhebox) [inline util/admin/admin_test.go:380]: Ditto.

**Review by xhebox — COMMENTED** [review]
Rest LGTM

**Comment by tisonkun** [comment]
@xhebox may we postpone the extraction work when we really have cases to be merged? I found `infoschema/perfschema/tables_tests.go` also get a mock store but setup / teardown differ quite a bit.

It is better we have a clear idea about how this potential testkit to be used, as "tk.MustExec", clear pattern. Otherwise we push too much to a first time contributor.

**Comment by xhebox** [comment]
@tisonkun sure, if you like.

**Review by xhebox — APPROVED** [review]

**Review by tiancaiamao — APPROVED** [review]

**Pull request merged**